src/testing/helpers.ts

```diff
@@ -0,0 +1,34 @@
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+// find a component within a debug element
+export function findComponent(el: DebugElement, tagName: string): any {
+  let subEl = el.query(By.css(tagName));
+  expect(subEl).not.toBeNull(`Unable to find component ${tagName}`);
+  return subEl.componentInstance;
+}
+
+// nicely stringify a debug/native element
+export function niceEl(el: DebugElement|any): string {
```

Comment:
There are probably libraries that could do this.  But since I needed to blacklist all the `_ngSomething` attributes anyways, and nuke all the angular-bindings-comments, I went with the custom magic-box approach.